test/files/neg/t7187-deprecation.scala

```diff
@@ -15,13 +15,13 @@ class EtaExpand214 {
   def m3(x: Int) = x
 
   val t1: () => Any  = m1   // error
-  val t2: () => Any  = m2   // eta-expanded with lint warning
-  val t2AcciSam: AcciSamZero = m2   // eta-expanded with lint warning + sam warning
-  val t2Sam: SamZero = m2   // eta-expanded with lint warning
+  val t2: () => Any  = m2   // eta-expanded, only warns w/ -Xlint:eta-zero
+  val t2AcciSam: AcciSamZero = m2   // eta-expanded, only warns w/ -Xlint:eta-zero or -Xlint:eta-sam
+  val t2Sam: SamZero = m2   // eta-expanded, only warns w/ -Xlint:eta-zero
   val t3: Int => Any = m3   // ok
 
   val t4 = m1 // apply
-  val t5 = m2 // apply, ()-insertion
+  val t5 = m2 // warn: apply, ()-insertion, will eta-expanded in 3.0
```

Comment:
I think that's not right, nilary methods don't eta-expand in 3.0, unless the expected type is a Function0.